lsp/src/Language/LSP/Server/Core.hs

```diff
@@ -710,31 +752,117 @@ flushDiagnosticsBySource maxDiagnosticCount msource = join $ stateState resDiagn
         case mdp of
           Nothing -> return ()
           Just params -> do
-            sendToClient $ J.fromServerNot $ J.TNotificationMessage "2.0" J.SMethod_TextDocumentPublishDiagnostics params
+            sendToClient $ L.fromServerNot $ L.TNotificationMessage "2.0" L.SMethod_TextDocumentPublishDiagnostics params
       in (act,newDiags)
 
 -- ---------------------------------------------------------------------
 
 -- | The changes in a workspace edit should be applied from the end of the file
 -- toward the start. Sort them into this order.
-reverseSortEdit :: J.WorkspaceEdit -> J.WorkspaceEdit
-reverseSortEdit (J.WorkspaceEdit cs dcs anns) = J.WorkspaceEdit cs' dcs' anns
+reverseSortEdit :: L.WorkspaceEdit -> L.WorkspaceEdit
+reverseSortEdit (L.WorkspaceEdit cs dcs anns) = L.WorkspaceEdit cs' dcs' anns
   where
     cs' :: Maybe (Map.Map Uri [TextEdit])
     cs' = (fmap . fmap ) sortTextEdits cs
 
-    dcs' :: Maybe [J.DocumentChange]
+    dcs' :: Maybe [L.DocumentChange]
     dcs' = (fmap . fmap) sortOnlyTextDocumentEdits dcs
 
-    sortTextEdits :: [J.TextEdit] -> [J.TextEdit]
-    sortTextEdits edits = L.sortOn (Down . (^. J.range)) edits
+    sortTextEdits :: [L.TextEdit] -> [L.TextEdit]
+    sortTextEdits edits = L.sortOn (Down . (^. L.range)) edits
 
-    sortOnlyTextDocumentEdits :: J.DocumentChange -> J.DocumentChange
-    sortOnlyTextDocumentEdits (J.InL (J.TextDocumentEdit td edits)) = J.InL $ J.TextDocumentEdit td edits'
+    sortOnlyTextDocumentEdits :: L.DocumentChange -> L.DocumentChange
+    sortOnlyTextDocumentEdits (L.InL (L.TextDocumentEdit td edits)) = L.InL $ L.TextDocumentEdit td edits'
       where
         edits' = L.sortOn (Down . editRange) edits
-    sortOnlyTextDocumentEdits (J.InR others) = J.InR others
+    sortOnlyTextDocumentEdits (L.InR others) = L.InR others
+
+    editRange :: L.TextEdit L.|? L.AnnotatedTextEdit -> L.Range
+    editRange (L.InR e) = e ^. L.range
+    editRange (L.InL e) = e ^. L.range
+
+--------------------------------------------------------------------------------
+-- CONFIG
+--------------------------------------------------------------------------------
 
-    editRange :: J.TextEdit J.|? J.AnnotatedTextEdit -> J.Range
-    editRange (J.InR e) = e ^. J.range
-    editRange (J.InL e) = e ^. J.range
+-- | Given a new config object, try to update our config with it.
+tryChangeConfig :: (m ~ LspM config) => LogAction m (WithSeverity LspCoreLog) -> J.Value -> m ()
+tryChangeConfig logger newConfigObject = do
+  parseCfg <- LspT $ asks resParseConfig
+  res <- stateState resConfig $ \oldConfig -> case parseCfg oldConfig newConfigObject of
+    Left err -> (Left err, oldConfig)
+    Right !newConfig -> (Right newConfig, newConfig)
```

Comment:
Any reason to use BangPattern here?

Reply:
Don't think so - it was there before. I'll drop it.